src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -2476,6 +2607,9 @@ ss::future<checked<tm_transaction, tx_errc>> tx_gateway_frontend::get_tx(
   ss::shared_ptr<tm_stm> stm,
   kafka::transactional_id tid,
   model::timeout_clock::duration timeout) {
+    if (!stm->hosts(tid)) {
+        co_return tx_errc::not_coordinator;
+    }
```

Comment:
Please don't modify `get_tx` & `get_ongoing_tx`, they are used in the core of the txn protocol and changing them is more risky that adding a check in the beginning of `do_end_txn`, `do_add_partition_to_tx`, `do_add_offsets_to_tx` and `do_init_tm_tx`. If you don't like the duplication you may introduced an overloaded `tm_stm::sync(kafka::transactional_id)` which checks `hosts` after `sync`